packages/deployments/contracts/contracts/messaging/connectors/arbitrum/ArbitrumHubConnector.sol

```diff
@@ -47,20 +97,47 @@ contract ArbitrumHubConnector is HubConnector {
     address _amb,
     address _rootManager,
     address _mirrorConnector,
-    uint256 _mirrorGas,
-    uint256 _defaultGasPrice,
-    address _outbox
-  ) HubConnector(_domain, _mirrorDomain, _amb, _rootManager, _mirrorConnector, _mirrorGas) {
-    defaultGasPrice = _defaultGasPrice;
+    address _outbox,
+    uint256 _maxSubmissionCostCap,
+    uint256 _maxGasCap,
+    uint256 _gasPriceCap
+  ) HubConnector(_domain, _mirrorDomain, _amb, _rootManager, _mirrorConnector) {
     outbox = IArbitrumOutbox(_outbox);
     rollup = IArbitrumRollup(outbox.rollup());
+
+    // Set initial caps for L1 -> L2 messages
+    maxSubmissionCostCap = _maxSubmissionCostCap;
```

Comment:
Is there a reason for these caps in the contract? Whoever calls the function will be funding with their own gas so I don't see the need to enforce a cap here.

Reply:
it depends on how the relayer fees are being repaid -- i think best practice for these things is to allow runtime info to be passed in up to a point, but theres always a risk relayer could bid up to the cap constantly.